Issue#50 introduce api test framework

Romanov-Niko opened the pull request:
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Created database and application creation in containers with docker-maven-plugin.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Created database and application creation in containers with docker-maven-plugin.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Moved database and application creation in a separate profile.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Moved database and application creation in a separate profile.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Added link to the database in container, renamed database container name in docker-compose and added phase of build and start in docker-maven-plugin.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker

Added link to the database in container, renamed database container name in docker-compose and added phase of build and start in docker-maven-plugin.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactor docker profile

Moved docker profile from home-application to home-api-tests.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactor docker profile

Moved docker profile from home-application to home-api-tests.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactored docker profile

Added properties(postgres, jar names and version, containers ports) and links to them.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactored docker profile

Added properties(postgres, jar names and version, containers ports) and links to them.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Removed hibernate ddl from properties

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Removed hibernate ddl from properties

PoLytvy committed:
Issue#50-added-test-class

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactored docker profile

Added new properties and links to them, moved maven dependency plugin into the docker profile, changed artifacts directory from target/ to target/docker/artifacts.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Refactored docker profile

Added new properties and links to them, moved maven dependency plugin into the docker profile, changed artifacts directory from target/ to target/docker/artifacts.

PoLytvy committed:
Issue#50: Fixed client pom.mustache file

Romanov-Niko committed:
Added new line at the end in docker compose

PoLytvy committed:
Issue#50: added NewsApiTest

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Random port and stop existing containers

Added plugin that allows us to use available ports and prevent collisions. Added stop goal to execute section of docker-maven-plugin to stop existing containers.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Random port and stop existing containers

Added plugin that allows us to use available ports and prevent collisions. Added stop goal to execute section of docker-maven-plugin to stop existing containers.

Romanov-Niko committed:
Merge remote-tracking branch 'origin/Issue#50-Deploy-infrastructure-on-docker' into Issue#50-Deploy-infrastructure-on-docker

# Conflicts:
#	docker-compose.yml

PoLytvy committed:
Issue#50: configured plagins in pom file

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added profile to github actions

Added docker profile to github actions and changed version from 0.34.1 to 0.24.0 to fix bug with stopping of build.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added profile to github actions

Added docker profile to github actions and changed version from 0.34.1 to 0.24.0 to fix bug with stopping of build.

PoLytvy committed:
Issue#50: Updated API tests, corrected openapi file

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added data migration module building in github actions

Added separate building of data migration module on github actions, cause there was problems with retrieving jar from unbuilt module with maven-dependency-plugin.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added data migration module building in github actions

Added separate building of data migration module on github actions, cause there was problems with retrieving jar from unbuilt module with maven-dependency-plugin.

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added dependency on data migration module

Fixed problem with maven dependency plugin, now we don't need to build data-migration module separately

Romanov-Niko committed:
Issue#50-Deploy-infrastructure-on-docker/Added dependency on data migration module

Fixed problem with maven dependency plugin, now we don't need to build data-migration module separately

PoLytvy committed:
Issue#104: Made test profile and finished delete test

Romanov-Niko committed:
Merge pull request #113 from ita-social-projects/Issue#50-Introduce-Api-tests

Issue#50 introduce api tests

Romanov-Niko committed:
Merge remote-tracking branch 'origin/Issue#50-Deploy-infrastructure-on-docker' into Issue#50-Introduce-Api-test-framework

Romanov-Niko committed:
Issue#50-Deploy-Infrastructure-on-docker/Added tests

Romanov-Niko committed:
Merge branch 'dev' of https://github.com/ita-social-projects/Home into Issue#50-Deploy-infrastructure-on-docker

Romanov-Niko committed:
Issue#50-Deploy-Infrastructure-on-docker/Added profile with tests to github actions

Romanov-Niko committed:
Merge pull request #101 from ita-social-projects/Issue#50-Deploy-infrastructure-on-docker

Issue#50 deploy infrastructure on docker

Romanov-Niko committed:
Merge branch 'Issue#50-Introduce-Api-test-framework' of https://github.com/ita-social-projects/Home into Issue#50-Introduce-Api-test-framework

 Conflicts:
	.github/workflows/build.yml
	.github/workflows/packaging.yml
	home-api-tests/pom.xml

Romanov-Niko committed:
Issue#50-Deploy-Infrastructure-on-docker/Added test classes

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -2,15 +2,313 @@
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
```
Please move this definition to the `<id>api-tests</id>` profile

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -2,15 +2,313 @@
+    <dependencies>
+        <dependency>
+            <groupId>com.softserveinc.ita.homeproject</groupId>
+            <artifactId>home-data-migration</artifactId>
```
I think it is mandatory to add the same for home-application

Romanov-Niko replied:
It makes conversion errors. For example `javax.ws.rs.core.Response cannot be converted to com.softserveinc.ita.homeproject.model.ReadN
ews`

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -2,15 +2,313 @@
-    <artifactId>home-api-tests</artifactId>
+    <profiles>
+        <profile>
+            <id>docker</id>
```
Let's rename this profile to `api-tests-infrastructure`

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/NewsApiIT.java:
```diff
@@ -0,0 +1,110 @@
+
+    private final NewsApi newsApi;
+
+    {
```
What is it about moving this code block to util class?

Romanov-Niko replied:
Done

kh0ma commented on home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/UserApiIT.java:
```diff
@@ -0,0 +1,105 @@
+        assertEquals(update.getFirstName(), updated.getFirstName());
+        assertEquals(update.getLastName(), updated.getLastName());
+    }
+}
```
`No newline`

Romanov-Niko replied:
Done

Romanov-Niko committed:
Issue#50-Deploy-Infrastructure-on-docker/Refactored home-api-tests module

Failsafe plugin was added to api-tests profile, created ApiClientUtil class with function that provides client, changed profiles in github actions.
